core/test/utils/assertions.hpp

```diff
@@ -999,6 +999,38 @@ ::testing::AssertionResult array_equal(const std::string& first_expression,
 }
 
 
+template <typename ValueType>
+::testing::AssertionResult segmented_array_equal(
```

Comment:
Can you also add a small test for this in `core/test/utils/assertions_test.cpp`  ?